inference-engine/tests/functional/inference_engine/ir_serialization/custom_ops.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <ie_api.h>
+#include <ie_iextension.h>
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "ie_core.hpp"
+#include "ngraph/ngraph.hpp"
+#include "transformations/serialize.hpp"
+
+#ifndef IR_SERIALIZATION_MODELS_PATH  // should be already defined by cmake
+#define IR_SERIALIZATION_MODELS_PATH ""
+#endif
+
+class SerializationTestOp : public ngraph::op::Op {
+public:
+    static constexpr ngraph::NodeTypeInfo type_info{"SerializationTestOp", 0};
+    const ngraph::NodeTypeInfo& get_type_info() const override {
+        return type_info;
+    }
+
+    SerializationTestOp() = default;
+    SerializationTestOp(const ngraph::Output<ngraph::Node>& arg, bool test1,
+                        int64_t test2)
+        : Op({arg}), test1(test1), test2(test2) {
+        constructor_validate_and_infer_types();
+    }
+
+    void validate_and_infer_types() override {
+        auto input_shape = get_input_partial_shape(0).to_shape();
+
+        ngraph::Shape output_shape(input_shape);
+        for (int i = 0; i < input_shape.size(); ++i) {
+            output_shape[i] = input_shape[i] * test2 + (test1 ? 0 : 1);
+        }
+
+        set_output_type(0, get_input_element_type(0),
+                        ngraph::PartialShape(output_shape));
+    }
+
+    std::shared_ptr<ngraph::Node> clone_with_new_inputs(
+        const ngraph::OutputVector& new_args) const override {
+        if (new_args.size() != 1) {
+            throw ngraph::ngraph_error("Incorrect number of new arguments");
+        }
+
+        return std::make_shared<SerializationTestOp>(new_args.at(0), test1,
+                                                     test2);
+    }
+
+    bool visit_attributes(ngraph::AttributeVisitor& visitor) override {
+        visitor.on_attribute("test1", test1);
+        visitor.on_attribute("test2", test2);
+        return true;
```

Comment:
can we use `template_extension` library for such needs?

Reply:
If we want to depend our test on template_extension library then sure, I will refactor tomorrow. In principal I prefer my tests be as independent as possible, hence this approach.

Reply:
I agree that we can use `template_extension` you can find here: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/inference_engine/extension.cpp#L332 how we already use it in IE tests.

Reply:
Done in this [comit](https://github.com/openvinotoolkit/openvino/pull/2862/commits/4981e0aa608fd493ce7164ca418dce6ee7749529?file-filters%5B%5D=.cpp)